nntrainer/layers/rnn.cpp

```diff
@@ -81,6 +85,21 @@ int RNNLayer::initialize(Manager &manager) {
   h_prev = Tensor(bias_dim);
   h_prev.setZero();
 
+  TensorDim d = input_dim[0];
+  d.width(unit);
+
+  // We do not need this if we reuse net_hidden[0]. But if we do, then the unit
+  // test will fail. Becuase it modifies the date during gradient calculation
+  // TODO : We could control with something like #define test to save memory
```

Comment:
agreed, we can define a macro from meson based on unit_test to disable some optimizations for unittests.